### PR TITLE
PEFT perf and TE spec fixes

### PR DESCRIPTION
**Pull request by JimmyZhang12**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/recipes/llama3_70b.py [hunk 1]
```diff
@@ -306,6 +306,7 @@ def finetune_recipe(
         recipe.peft.dim = 16
         recipe.peft.alpha = 32
         recipe.peft.target_modules = ['linear_qkv']
+        recipe.optim.config.use_distributed_optimizer = False
```
Comment (cuichenx) [inline nemo/collections/llm/recipes/llama3_70b.py:309]: is this because the extra communication makes distributed optimizer slower for lora?

File: nemo/lightning/pytorch/callbacks/peft.py [hunk 2]
```diff
@@ -91,8 +91,13 @@ def __call__(self, model: nn.Module) -> nn.Module:
         Returns:
             nn.Module: The transformed model with PEFT applied.
         """
-        self.freeze_model(model)
-        model.walk(self.transform)
+        if len(model) > 1:
+            for model_chunk in model:
+                model_chunk.freeze()
+                model_chunk.walk(self.transform)
```
Comment (cuichenx) [inline nemo/lightning/pytorch/callbacks/peft.py:97]: is this for PP? could you add a comment here please

File: nemo/collections/llm/recipes/llama3_70b.py (hunk 1 above)
Comment (JimmyZhang12) [reply nemo/collections/llm/recipes/llama3_70b.py:309]: yes, there is also some memory overhead, and since the number of params is tiny, we can just disable

**Review by erhoo82 — APPROVED** [review]
